speech/cloud-client/transcribe.py

```diff
@@ -57,13 +55,48 @@ def main(speech_file):
     # [END send_request]
 
 
+def transcribe_gcs(gcs_uri):
+    """Transcribes the audio file specified by the gcs_uri."""
+    # [START authenticating_gcs]
+    # Application default credentials provided by env variable
+    # GOOGLE_APPLICATION_CREDENTIALS
+    from google.cloud import speech
+    speech_client = speech.Client()
+    # [END authenticating]
+
+    # [START construct_request_gcs]
+    audio_sample = speech_client.sample(
+        None,
+        source_uri=gcs_uri,
+        encoding='FLAC',
+        sample_rate=16000)
+    # [END construct_request_gcs]
+
+    # [START send_request_gcs]
+    alternatives = speech_client.speech_api.sync_recognize(audio_sample)
+    for alternative in alternatives:
+        print('Transcript: {}'.format(alternative.transcript))
+    # [END send_request_gcs]
+
+
+def main(path):
```

Comment:
We generally don't put main functions in our samples, just place this logic below or do separate subparsers for local path and gcs path.